### ISSUE TITLE
Segmentation fault, when sending email with autocrypt enabled

### ISSUE THREAD

**Issue by dngray** [neomutt/neomutt#2012]
When setting autocrypt on and trying to send an email I get a segmentation fault:

```
set autocrypt = yes
set autocrypt_dir = "$XDG_CONFIG_HOME/mutt/autocrypt"
bind index z autocrypt-acct-menu
```

```
GNU gdb (GDB) 8.3.1
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from neomutt...
Reading symbols from /usr/lib/debug/usr/bin/neomutt.debug...
(gdb) run
Starting program: /usr/bin/neomutt 
/usr/lib/../share/gcc-9.2.0/python/libstdcxx/v6/xmethods.py:731: SyntaxWarning: list indices must be integers or slices, not str; perhaps you missed a comma?
  refcounts = ['_M_refcount']['_M_pi']
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[Detaching after fork from child process 93052]
[Detaching after fork from child process 93057]
[Detaching after fork from child process 93059]
[Detaching after fork from child process 93061]
[Detaching after fork from child process 93063]
[Detaching after fork from child process 93065]
[Detaching after fork from child process 93076]
double free or corruption (out)

Program received signal SIGABRT, Aborted.
0x00007ffff758cf25 in raise () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff758cf25 in raise () at /usr/lib/libc.so.6
#1  0x00007ffff7576897 in abort () at /usr/lib/libc.so.6
#2  0x00007ffff75d0258 in __libc_message () at /usr/lib/libc.so.6
#3  0x00007ffff75d777a in  () at /usr/lib/libc.so.6
#4  0x00007ffff75d95d0 in _int_free () at /usr/lib/libc.so.6
#5  0x00000001001588e1 in mutt_mem_free (ptr=0x7fffffff81c0) at mutt/memory.c:75
#6  0x00000001001480c2 in mutt_addr_free (ptr=0x7fffffff8118) at address/address.c:446
#7  0x0000000100149df2 in mutt_addrlist_clear (al=0x7fffffff8190) at address/address.c:1390
#8  0x00000001000c5a44 in mutt_autocrypt_ui_recommendation (e=0x10032be10, keylist=0x0)
    at autocrypt/autocrypt.c:662
#9  0x000000010002f7b3 in update_crypt_info (rd=0x7fffffff8360) at compose.c:504
#10 0x0000000100031423 in mutt_compose_menu (e=0x10032be10, fcc=0x7fffffff9a00, e_cur=0x0, flags=0)
    at compose.c:1134
#11 0x00000001000afd3d in ci_send_message (flags=0, e_templ=0x10032be10, tempfile=0x0, ctx=0x1001feac0, el=0x0)
    at send.c:2424
#12 0x000000010005dea3 in mutt_index_menu () at index.c:3337
#13 0x0000000100071023 in main (argc=1, argv=0x7fffffffd258, envp=0x7fffffffd268) at main.c:1260
```

## NeoMutt Version

```
NeoMutt 20191129
Copyright (C) 1996-2016 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.3.13.a-1-hardened (x86_64)
ncurses: ncurses 6.1.20180127 (compiled with 6.1.20180127)
libidn2: 2.3.0 (compiled with 2.3.0)
GPGme: 1.13.1
hcache backends: kyotocabinet, gdbm, lmdb

Compiler:
Using built-in specs.
COLLECT_GCC=cc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++,d --enable-shared --enable-threads=posix --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --enable-checking=release --enable-default-pie --enable-default-ssp --enable-cet=auto gdc_include_dir=/usr/include/dlang/gdc
Thread model: posix
gcc version 9.2.0 (GCC) 

Configure options: --prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --gpgme --enable-lua --notmuch --gss --ssl --sasl --with-ui=ncurses --with-idn2=/usr --with-sqlite=/usr --autocrypt --idn=0 --idn2=1 --lmdb --kyotocabinet --gdbm

Compilation CFLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -g -O0 -fvar-tracking-assignments -fdebug-prefix-map=/home/david/git/d18/pkgbuilds/neomutt/src=/usr/src/debug -std=c99 -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo -gnutls +gpgme +gss +hcache -homespool +idn +inotify 
  -locales_hack +lua +meta -mixmaster +nls +notmuch +openssl +pgp +sasl +smime 
  +sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version Archlinux
* Were you using multiple copies of NeoMutt at once? no
* Were you using 'screen' or 'tmux'? no
* Is your email local (maildir) or remote (IMAP)? maildir

**Comment by dngray** [comment]
This appears to happen when you've got no keys in your autocrypt directory. It would appear the auto crypt plugin cannot import them from gpg, and you must manually copy them from ~/.gnupg

**Comment by flatcap** [comment]
fixed in 72298a4fb
